fdbserver/LogRouter.actor.cpp

```diff
@@ -304,48 +305,68 @@ ACTOR Future<Void> waitForVersion(LogRouterData* self, Version ver) {
 	return Void();
 }
 
+ACTOR Future<Reference<ILogSystem::IPeekCursor>> getPeekCursorData(LogRouterData* self,
+                                                                   Reference<ILogSystem::IPeekCursor> r,
+                                                                   Version startVersion) {
+	state Reference<ILogSystem::IPeekCursor> result = r;
+	state bool useSatellite = SERVER_KNOBS->LOG_ROUTER_PEEK_FROM_SATELLITES_PREFERRED;
+	loop {
+		Future<Void> getMoreF = Never();
+		if (result) {
+			getMoreF = result->getMore(TaskPriority::TLogCommit);
+			++self->getMoreCount;
+			if (!getMoreF.isReady()) {
+				++self->getMoreBlockedCount;
+			}
+		}
+		state double startTime = now();
+		choose {
+			when(wait(getMoreF)) {
+				double peekTime = now() - startTime;
+				self->peekLatencyDist->sampleSeconds(peekTime);
+				self->getMoreTime += peekTime;
+				self->maxGetMoreTime = std::max(self->maxGetMoreTime, peekTime);
+				return result;
+			}
+			when(wait(self->logSystemChanged)) { // FIXME: does this actually happen?
```

Comment:
I am not sure if this FIXME is still valid. If `result` is null, to my understanding this `when` is the only branch that updates the `result` to a real PeekCursor object. Maybe it should be removed?